apps/dotcom/client/e2e/fixtures/tla-test.ts

```diff
@@ -1,34 +1,139 @@
-		testUse(new Sidebar(page))
+		await testUse(new Sidebar(page))
 	},
 	editor: async ({ page, sidebar }, testUse) => {
-		testUse(new Editor(page, sidebar))
+		await testUse(new Editor(page, sidebar))
 	},
 	homePage: async ({ page, editor }, testUse) => {
-		testUse(new HomePage(page, editor))
+		await testUse(new HomePage(page, editor))
 	},
 	database: async ({ page }, testUse) => {
-		testUse(new Database(page))
+		await testUse(new Database(page, test.info().parallelIndex))
 	},
 	shareMenu: async ({ page }, testUse) => {
-		testUse(new ShareMenu(page))
+		await testUse(new ShareMenu(page))
 	},
+	deleteFileDialog: async ({ page }, testUse) => {
+		await testUse(new DeleteFileDialog(page))
+	},
+	// This is an auto fixture which makes sure that we are on the home page when the test starts
+	// and that we clean up when the tests completes
+	setupAndCleanup: [
+		async ({ page, homePage, database }, testUse) => {
+			// All the code before testUse is run before each test
+
+			// Make sure we don't get marked as a bot
+			// https://clerk.com/docs/testing/playwright/overview
+			await setupClerkTestingToken({ page })
+
+			await homePage.goto()
+			await homePage.isLoaded()
+
+			await testUse()
+
+			// All the code after testUse is run after each test
+			await database.reset()
+		},
+		{ auto: true },
+	],
+	storageState: ({ workerStorageState }, testUse) => testUse(workerStorageState),
+	workerStorageState: [
+		async ({ browser }, testUse) => {
+			const id = test.info().parallelIndex
+
+			const fileName = getStorageStateFileName(id, 'huppy')
+			if (fs.existsSync(fileName)) {
+				// Reuse existing authentication state if any.
+				await testUse(fileName)
+				return
+			}
+			for (const user of ['huppy' as const, 'suppy' as const]) {
+				const fileName = getStorageStateFileName(id, user)
+				// Important: make sure we authenticate in a clean environment by unsetting storage state.
+				const page = await browser.newPage({ storageState: undefined })
+				let email: string
+				if (user === 'huppy') {
+					email = USERS[id]
+				} else {
+					email = OTHER_USERS[id]
+				}
+				const sidebar = new Sidebar(page)
+				const editor = new Editor(page, sidebar)
+				const homePage = new HomePage(page, editor)
+				await homePage.loginAs(email)
+				await expect(page.getByTestId('tla-sidebar-layout')).toBeVisible()
+
+				await page.context().storageState({ path: fileName })
+				await page.close()
+			}
+			await testUse(fileName)
+		},
+		{ scope: 'worker' },
+	],
 })
 
 export { expect } from '@playwright/test'
+
+export function step(name: string) {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
+	return function (target: Function, context: any) {
+		if (context.kind === 'method') {
+			return async function (...args: any[]) {
+				return await test.step(name, async () => {
+					// @ts-expect-error Parameter 'this' implicitly has an 'any' type.ts(7006)
+					return target.apply(this, args)
+				})
```

Comment:
Love this decorator. I think you can infer the step names and just put `@step` without arguments everywhere.
```suggestion
				return await test.step(`${this.constructor.name}.${context.name}`, async () => {
					// @ts-expect-error Parameter 'this' implicitly has an 'any' type.ts(7006)
					return target.apply(this, args)
				})
```